### DISCUSSION TITLE
[Question] Pushing gauge metric with label

### DISCUSSION THREAD

**Discussion by chardch** [prometheus/client_python#910] [General]
I was trying to follow https://github.com/prometheus/client_python#exporting-to-a-pushgateway, but adding a label to my metric, something like

```
from prometheus_client import CollectorRegistry, Gauge, push_to_gateway

registry = CollectorRegistry()
g = Gauge('job_last_success_unixtime', 'Last time a batch job successfully finished', labelnames=["mylabel"], registry=registry)
g.set_to_current_time()
g.labels("mylabel="testing").set(63.3)
push_to_gateway('localhost:9091', job='batchA',  registry=registry)
```

With this, I was getting this error `ValueError: gauge metric is missing label values`. It looks like in [1] that we can't have labels when calling `set_to_current_time()`? Is that correct? The documentation seems to always suggest calling `set_to_current_time()` for gauges, but it's not clear to me what purpose it serves. Regardless though, I would ideally like to track when the metric is recorded and use it with labels. Any help on how to do this would be appreciated. Thanks!

[1] https://github.com/prometheus/client_python/blob/master/prometheus_client/metrics.py#L81

**Comment by chardch** [comment]
Realized the issue was that `g.set_to_current_time()` should be `g.labels("mylabel="testing").set_to_current_time()` as well to make this work

**Comment by SunnyKrJha** [comment]
registry = CollectorRegistry()
g = Gauge('my_metric', 'Description of my_metric', labelnames=['ln'], registry=registry)
g.labels(ln='labelvalue').set(32)

For the above snippet of Code, I get the following error - 
    raise UnicodeError("label empty or too long")
    label empty or too long 

Can some one help with how to set the labels correctly with Gauge ?

Also, if the labels are that critical for correct functioning, I believe that README.md should reflect them.
